lib/stdlib/src/maps.erl

```diff
@@ -497,6 +498,48 @@ with_1([K|Ks], Map) ->
     end;
 with_1([], _Map) -> [].
 
+%% groups_from_list/2 & groups_from_list/3
+
+-spec groups_from_list(Fun, List) -> MapOut when
+    Fun :: fun((Elem :: T) -> Selected),
+    MapOut :: #{Selected => List},
+    Selected :: term(),
+    List :: [T],
+    T :: term().
+
+groups_from_list(_Fun, []) ->
+    #{};
+groups_from_list(Fun, List) when is_function(Fun, 1) ->
+    groups_from_list_1(Fun, lists:reverse(List), #{}).
```

Comment:
In order to support extended error information, the following changes are necessary:

```suggestion
groups_from_list(Fun, List0) when is_function(Fun, 1) ->
    try lists:reverse(List0) of
        List ->
            groups_from_list_1(Fun, List, #{})
    catch
        error:_ ->
            badarg_with_info([Fun, List0])
    end;
groups_from_list(Fun, List) ->
    badarg_with_info([Fun, List]).
```